backend/core/database/repositories/blog.py

```diff
@@ -0,0 +1,85 @@
+from typing import Self, Type, Any, Sequence
+
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import async_sessionmaker
+
+from core.database.repositories.base import BaseRepository
+from core.database import engine
+from core.database.models.blog import BlogORM
+
+
+class BlogRepository(BaseRepository):
+    def __init__(self):
+        self.session: async_sessionmaker
+
+    async def __aenter__(self: Self) -> Self:
+        self.session = async_sessionmaker(engine)
+        return self
+
+    async def __aexit__(self, exc_type, exc_value, exc_tb) -> None:  # noqa
+        return await self.session().close()
```

Comment:
Неправильно закрываем сессию
```suggestion
    async def __aexit__(self, exc_type, exc_value, exc_tb) -> None:  # noqa
        return await self.session.close()
```